syntax = "proto3";

import 'pw_protobuf_protos/common.proto';

package chip.rpc;

// Define the message for a synchronized end device with necessary fields
message SynchronizedDevice {
  uint64 node_id = 1;
}

message KeepActiveChanged {
  uint64 node_id = 1;
  uint32 promised_active_duration = 2;
}

service FabricBridge {
  rpc AddSynchronizedDevice(SynchronizedDevice) returns (pw.protobuf.Empty){}
  rpc RemoveSynchronizedDevice(SynchronizedDevice) returns (pw.protobuf.Empty){}  
  rpc ActiveChanged(KeepActiveChanged) returns (pw.protobuf.Empty){}  
}
